pkg/reconciler/instances/istio/action.go

```diff
@@ -1,410 +1,132 @@
-	if componentVersion == "" {
-		return true, nil
-	}
-	componentHelperVersion, err := newHelperVersionFrom(componentVersion)
-	if err != nil {
-		return false, err
-	}
-	targetHelperVersion, err := newHelperVersionFrom(targetVersion)
-	if err != nil {
-		return false, err
-	}
 
-	componentVsTargetComparison := targetHelperVersion.compare(componentHelperVersion)
-	if !amongOneMinor(componentHelperVersion, targetHelperVersion) {
-		return false, fmt.Errorf("could not perform %s for %s from version: %s to version: %s - the difference between versions exceed one minor version",
-			getActionTypeFrom(componentVsTargetComparison), componentName, componentVersion, targetVersion)
-	}
-
-	return true, nil
-}
-
-func getActionTypeFrom(comparison int) string {
-	switch comparison {
-	case 1:
-		return "upgrade"
-	case 0:
-		return "reconciliation"
-	case -1:
-		return "downgrade"
-	default:
-		return "unknown"
-	}
-}
-
-func amongOneMinor(first, second helperVersion) bool {
-	return first.ver.Major == second.ver.Major && (first.ver.Minor == second.ver.Minor || first.ver.Minor-second.ver.Minor == -1 || first.ver.Minor-second.ver.Minor == 1)
-}
-
-func unDeployIstioRelatedResources(context context.Context, manifest string, client kubernetes.Client, logger *zap.SugaredLogger) error {
-	logger.Debugf("Undeploying istio related dashboards")
-	// multiple calls necessary, please see: https://github.com/kyma-incubator/reconciler/issues/367
-	_, err := client.Delete(context, manifest, "kyma-system")
-	if err != nil {
-		return err
-	}
-	logger.Debugf("Undeploying other istio related resources")
-	_, err = client.Delete(context, manifest, istioNamespace)
-	if err != nil {
-		return err
-	}
-
-	return nil
-}
-
-type envoyAction struct {
-	getIstioPerformer bootstrapIstioPerformer
+	return retry.Do(func() error {
+		return checkIfIstioIsReady(context, k8sClient)
+	}, retry.Attempts(retryAttempts), retry.Delay(retryDelay))
 }
 
-func NewEnvoyAction(getIstioPerformer bootstrapIstioPerformer) service.Action {
-	return &envoyAction{
-		getIstioPerformer: getIstioPerformer,
-	}
-}
-
-func (e *envoyAction) Run(context *service.ActionContext) error {
-	context.Logger.Debug("Istio envoy filter apply action triggered")
-
-	provider := clientset.DefaultProvider{}
-
-	istioClient, err := provider.GetIstioClient(context.KubeClient.Kubeconfig())
+func checkIfIstioIsReady(context *service.ActionContext, k8sClient client.Client) error {
+	var istioList v1alpha1.IstioList
+	err := k8sClient.List(context.Context, &istioList)
 	if err != nil {
 		return err
 	}
-
-	present, err := envoy.IsEnvoyFilterPresent(context.Context, istioClient)
-	if err != nil {
-		return err
-	}
-
-	if !present {
-		err := envoy.CreateEnvoyFilter(context.Context, istioClient)
-		if err != nil {
-			return err
-		}
-
-		err = ingressgateway.RestartDeployment(context.Context, istioClient)
-		if err != nil {
-			return err
+	for _, item := range istioList.Items {
+		if item.Status.State == "Ready" {
```

Comment:
We should discuss what to do when customer tries to delete the Istio module.
Should the check also consider Istio in `Warning` state? Maybe we should end reconciliation with warning as well?